### PR TITLE
Ignore virtual members

### PR DESCRIPTION
**Pull request by MichalJankowskii**
Added possibility to ignore virtual members generation - it speeds up object creation when this feature is turned on

### CODE REVIEW

File: src/Objectivity.AutoFixture.XUnit2.AutoMoq/SpecimenBuilder/IgnoreVirtualMembersSpecimenBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var pi = request as PropertyInfo;
+            if (pi == null)
+            {
+                return new NoSpecimen();
+            }
+
+            if (pi.GetGetMethod().IsVirtual)
```
Comment (piotrzajac) [inline src/Objectivity.AutoFixture.XUnit2.AutoMoq/SpecimenBuilder/IgnoreVirtualMembersSpecimenBuilder.cs]: Consider combining into single if statement: if (pi != null && pi.GetGetMethod().IsVirtual)

File: src/Objectivity.AutoFixture.XUnit2.AutoMoq/SpecimenBuilder/IgnoreVirtualMembersSpecimenBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+{
+    using System;
+    using System.Reflection;
+    using Ploeh.AutoFixture.Kernel;
+
+    internal class IgnoreVirtualMembersSpecimenBuilder : ISpecimenBuilder
+    {
+        public object Create(object request, ISpecimenContext context)
+        {
+            if (context == null)
```
Comment (piotrzajac) [inline src/Objectivity.AutoFixture.XUnit2.AutoMoq/SpecimenBuilder/IgnoreVirtualMembersSpecimenBuilder.cs]: Should we check/throw if we do not use context at all? When looking into AutoFixture code they do not throw at all.

File: src/Objectivity.AutoFixture.XUnit2.AutoMoq.Tests/SpecimenBuilder/IgnoreVirtualMembersSpecimenBuilderTests.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    public class IgnoreVirtualMembersSpecimenBuilderTests
+    {
+        private readonly ISpecimenContext context;
+
+        public IgnoreVirtualMembersSpecimenBuilderTests()
+        {
+            this.context = new Mock<ISpecimenContext>().Object;
+        }
+
+        [Fact]
```
Comment (piotrzajac) [inline src/Objectivity.AutoFixture.XUnit2.AutoMoq.Tests/SpecimenBuilder/IgnoreVirtualMembersSpecimenBuilderTests.cs]: Please add DisplayName in all newly introduced facts.

**Comment by piotrzajac** [comment]
There is already a new version waiting for you. Please take a look into my commit comments.
To introduce your change it is enough to introduce another customization with bool parameter in the constructor passed from optional attribute property, e.g.:

        public override IEnumerable<object[]> GetData(MethodInfo testMethod)
        {
            this.Fixture.Customize(new AutoMoqDataCustomization());
            this.Fixture.Customize(new IgnoreVirtualMembersCustomization(this.IgnoreVirtualMembers));

            return this.Provider.GetAttribute(this.Fixture, this.values).GetData(testMethod);
        }

That would be especially useful in MemberAutoMoqDataAttribute where we have no control over attributes any more but only on fixture itself.

File: src/Objectivity.AutoFixture.XUnit2.AutoMoq/Attributes/AutoMoqDataAttribute.cs [hunk 4]
```diff
@@ -15,16 +15,18 @@ namespace Objectivity.AutoFixture.XUnit2.AutoMoq.Attributes
...
     {
-        public AutoMoqDataAttribute()
-            : this(new Fixture())
+        public AutoMoqDataAttribute(bool ignoreVirtualMembers = false)
+            : this(new Fixture(), ignoreVirtualMembers)
         {
         }
 
-        public AutoMoqDataAttribute(IFixture fixture)
+        public AutoMoqDataAttribute(IFixture fixture, bool ignoreVirtualMembers = false)
```
Comment (MichalJankowskii) [inline src/Objectivity.AutoFixture.XUnit2.AutoMoq/Attributes/AutoMoqDataAttribute.cs]: Ok for MemberAutoMoqDataAttribute. Other cases I planned to discuss with you but I can see that now there is nothing to discuss :).

File: src/Objectivity.AutoFixture.XUnit2.AutoMoq/SpecimenBuilder/IgnoreVirtualMembersSpecimenBuilder.cs (hunk 2 above)
Comment (MichalJankowskii) [inline src/Objectivity.AutoFixture.XUnit2.AutoMoq/SpecimenBuilder/IgnoreVirtualMembersSpecimenBuilder.cs]: You are right. It is remnant from code prepared for previous version of AutoFixture.

File: src/Objectivity.AutoFixture.XUnit2.AutoMoq/SpecimenBuilder/IgnoreVirtualMembersSpecimenBuilder.cs (hunk 1 above)
Comment (MichalJankowskii) [inline src/Objectivity.AutoFixture.XUnit2.AutoMoq/SpecimenBuilder/IgnoreVirtualMembersSpecimenBuilder.cs]: In this case I would leave my implementation. It is easier to understand what this code is doing.

File: src/Objectivity.AutoFixture.XUnit2.AutoMoq.Tests/SpecimenBuilder/IgnoreVirtualMembersSpecimenBuilderTests.cs (hunk 3 above)
Comment (MichalJankowskii) [inline src/Objectivity.AutoFixture.XUnit2.AutoMoq.Tests/SpecimenBuilder/IgnoreVirtualMembersSpecimenBuilderTests.cs]: Yes of course.

**Comment by MichalJankowskii** [comment]
Ok. I will adjust my code.

**Comment by MichalJankowskii** [comment]
I committed code according to your suggestions but I am not 100% content about it. We can talk about it tomorrow.

**Comment by piotrzajac** [comment]
Looks much better than previous one. Let's discuss the details tomorrow.

**Pull request merged**